module/VuFind/src/VuFind/OAI/Server.php

```diff
@@ -444,24 +445,18 @@ protected function attachNonDeleted(
         $headerOnly = false,
         $set = ''
     ) {
-        // Get the XML (and display an error if it is unsupported):
         if ($format === false) {
-            $xml = '';      // no metadata if in header-only mode!
-        } elseif ('oai_vufind_json' === $format && $this->supportsVuFindMetadata()) {
-            $xml = $this->getVuFindMetadata($record);   // special case
-        } else {
-            $xml = $record
-                ->getXML($format, $this->baseHostURL, $this->recordLinkerHelper);
-            if ($xml === false) {
-                return false;
-            }
+            // If no format was requested, report success without doing anything:
+            return true;
         }
 
-        // Headers should be returned only if the metadata format matching
-        // the supplied metadataPrefix is available.
-        // If RecordDriver returns nothing, skip this record.
-        if (empty($xml)) {
-            return true;
+        $xml = $this->getRecordAsXML($record, $format);
+
+        // If returned XML is false, an error was encountered during the process
+        // of generating the XML file. Return false for calling functions, so the
+        // error can be processed properly.
```

Comment:
I think the existing comment remains valid and should be combined with the new one, like:

```suggestion
        // Headers should be returned only if the metadata format matching
        // the supplied metadataPrefix is available.
        // If returned XML is empty, return true to simply skip this record.
        // If returned XML is false, an error was encountered during the process
        // of generating the XML file. Return false for calling functions, so the
        // error can be processed properly.
```

Reply:
Adjusted this one also with minor change to my part. Also added a BC return for oai_vufind_json as it defaulted to true and thought it could be better not to change the old functionality.